cpp/ql/src/definitions.qll

```diff
@@ -156,7 +178,9 @@ Top definitionOf(Top e, string kind) {
     ) or (
       // type mention -> type
       kind = "T" and
+      TypeMentions::isCanonical(e) and
       e.(TypeMention).getMentionedType() = result and
+      not result instanceof ClassTemplateInstantiation and
```

Comment:
This line seems unrelated.

Reply:
You're right. It's morally part of #223, but I had somehow neglected to stage it. @jbj, for info -- let's keep this hotfix patch together, as it's what I tested.